#!/usr/bin/env python
PACKAGE = "controls"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#super important: these input types need to be the EXACT SAME as the order they
#appear in the PidAxis.hpp inputs_ vector definition. Otherwise it won't work.
input_type_enum = gen.enum([  gen.const("IMU_POS",     int_t, 0, "IMU_POS"),
                              gen.const("IMU_ACCEL",   int_t, 1, "IMU_ACCEL"),
                              gen.const("DEPTH",       int_t, 2, "DEPTH"),
                              gen.const("CAM_FRONT",   int_t, 3, "Front Camera Input"),
                              gen.const("CAM_BOTTOM",  int_t, 4, "Bottom camera input"),
                              gen.const("LOCALIZE",    int_t, 5, "Localization input")],
                              "Enumerates input type")

gen.add("Kp", double_t, 0, "Kp", 0.1, -5, 5)
gen.add("Ki", double_t, 0, "Ki", 0.1, -5, 5)
gen.add("Kd", double_t, 0, "Kd", 0.1, -5, 5)

gen.add("Save", bool_t, 0, "Save to yaml file", False)
gen.add("restoreDefaults", bool_t, 0, "Restore the vales saved in map", False)
gen.add("inputType", int_t, 0, "input used for config", 0, 0, 5, edit_method=input_type_enum)

exit(gen.generate(PACKAGE, "controls", "Pid"))
